aws/resource_aws_codebuild_project.go

```diff
@@ -1516,84 +1512,6 @@ func resourceAwsCodeBuildProjectArtifactsHash(v interface{}) int {
 	return hashcode.String(buf.String())
 }
 
-func resourceAwsCodeBuildProjectEnvironmentHash(v interface{}) int {
-	var buf bytes.Buffer
-	m := v.(map[string]interface{})
-
-	environmentType := m["type"].(string)
-	computeType := m["compute_type"].(string)
-	image := m["image"].(string)
-	privilegedMode := m["privileged_mode"].(bool)
-	imagePullCredentialsType := m["image_pull_credentials_type"].(string)
-	environmentVariables := m["environment_variable"].([]interface{})
-	buf.WriteString(fmt.Sprintf("%s-", environmentType))
-	buf.WriteString(fmt.Sprintf("%s-", computeType))
-	buf.WriteString(fmt.Sprintf("%s-", image))
-	buf.WriteString(fmt.Sprintf("%t-", privilegedMode))
-	buf.WriteString(fmt.Sprintf("%s-", imagePullCredentialsType))
-	if v, ok := m["certificate"]; ok && v.(string) != "" {
-		buf.WriteString(fmt.Sprintf("%s-", v.(string)))
-	}
-	if v, ok := m["registry_credential"]; ok && len(v.([]interface{})) > 0 && v.([]interface{})[0] != nil {
-		m := v.([]interface{})[0].(map[string]interface{})
-
-		if v, ok := m["credential"]; ok && v.(string) != "" {
-			buf.WriteString(fmt.Sprintf("%s-", v.(string)))
-		}
-
-		if v, ok := m["credential_provider"]; ok && v.(string) != "" {
-			buf.WriteString(fmt.Sprintf("%s-", v.(string)))
-		}
-	}
-	for _, e := range environmentVariables {
-		if e != nil { // Old statefiles might have nil values in them
-			ev := e.(map[string]interface{})
-			buf.WriteString(fmt.Sprintf("%s:", ev["name"].(string)))
-			// type is sometimes not returned by the API
-			if v, ok := ev["type"]; ok {
-				buf.WriteString(fmt.Sprintf("%s:", v.(string)))
-			}
-			buf.WriteString(fmt.Sprintf("%s-", ev["value"].(string)))
-		}
-	}
-
-	return hashcode.String(buf.String())
-}
-
-func resourceAwsCodeBuildProjectSourceHash(v interface{}) int {
```

Comment:
Since `auth` was not used in the hash calculation in this function, this should fix #6129 as well.